updated data retention docs (#4358)
segmentio/segment-docs: src/privacy/user-deletion-and-suppression.md

## Data retention

> Segment stores a copy of all event data received in Segment’s secure event Archives, on S3. By default, all workspaces are set to store data for an unlimited period of time, but you can modify the lifecycle policies for the data stored internally. We use this data for [data replays](/docs/guides/what-is-replay/) and for troubleshooting purposes.

Reviewer: ```suggestion
Segment stores a copy of all event data received in Segment’s secure event Archives, on S3. By default, all workspaces are set to store data for an unlimited period of time, but you can modify the lifecycle policies for the data stored internally. Segment uses this data for [data replays](/docs/guides/what-is-replay/) and for troubleshooting purposes.
```

> Segment recommends keeping at least 30 days of data to enable [replays](/docs/guides/what-is-replay/) of your data.

Reviewer: @muhammadatifjav do you mean 30 days worth of data, or keeping your data for 30 days?

Author: I'm not sure if there is a difference? We recommend keeping retention period of 30 days as mentioned in app as well.

Reviewer: ```suggestion
Segment recommends keeping your data for at least 30 days to enable [replays](/docs/guides/what-is-replay/) of your data.
```

> Segment recommends keeping at least 30 days of data to enable [Replays](/docs/guides/what-is-replay/) of your data.

Reviewer: ```suggestion
Segment recommends keeping at least 30 days of data to enable [replays](/docs/guides/what-is-replay/) of your data.
```

> To view the data retention settings, navigate to **Privacy > Settings > Data Retention** in Segment.

Reviewer: ```suggestion
To change your data retention settings, navigate to **Privacy > Settings > Data Retention** in Segment.
```
